### PR TITLE
feat: dislike blog post

### PR DESCRIPTION
**Pull request by traderstechie**
## Description:
A feature that allows users to react to a blog post with a dislike. It's a protected endpoint at `/api/v1/blogs/:{blog_id}/dislike`. It allows only authenticated users to dislike a blog post once.

## Related Issue:

[[FEAT]: Endpoint To Dislike A Blog Post #275](https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/issues/275)

## Motivation and Context
This is necessary, in order to enable blog writers to understand how readers perceive their posts.

## How Has This Been Tested?

- [x] Unittest with pytest.

## Types of changes

- [x] New feature (non-breaking change which adds functionality)

## Checklist:
 
- [x]  My code follows the code style of this project.
- [x]  I have updated the documentation accordingly.
- [x]  I have read the CONTRIBUTING document.
- [x]   I have added tests to cover my changes.
- [x]  All new and existing tests passed.

### CODE REVIEW

File: api/v1/routes/blog.py
Comment (joboy-dev) [inline api/v1/routes/blog.py:1]: There's a services file for blogs. Use that to make database queries. Don't do it in routes

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Make changes

**Review by joboy-dev — APPROVED** [review]

File: api/v1/routes/blog.py [hunk 1]
```diff
@@ -37,6 +39,74 @@ def get_all_blogs(db: Session = Depends(get_db)):
...
+def dislike_blog_post(
+    blog_id: str,
+    db: Session = Depends(get_db),
+    current_user: User = Depends(user_service.get_current_user)
+    ):
+    
+    try:
+        blog_service = BlogService(db)
+
+        if not current_user:
```
Comment (joboy-dev) [inline api/v1/routes/blog.py:52]: No need for this

File: api/v1/routes/blog.py [hunk 2]
```diff
@@ -37,6 +39,74 @@ def get_all_blogs(db: Session = Depends(get_db)):
...
 
 
+@blog.put("/{blog_id}/dislike", response_model=BlogDislikeResponse)
+def dislike_blog_post(
+    blog_id: str,
+    db: Session = Depends(get_db),
+    current_user: User = Depends(user_service.get_current_user)
+    ):
+    
+    try:
```
Comment (joboy-dev) [inline api/v1/routes/blog.py:49]: No need for a try and except block

File: api/v1/routes/blog.py
Comment (joboy-dev) [inline api/v1/routes/blog.py:1]: Any database query had to be done in services

File: requirements.txt [hunk 3]
```diff
@@ -61,6 +62,5 @@ typing_extensions==4.12.2
 urllib3==2.2.2
 uuid7==0.1.0
 uvicorn==0.30.3
-uvloop==0.19.0
```
Comment (joboy-dev) [inline requirements.txt:64]: Why did you remove this

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Make changes

**Comment by traderstechie** [comment]
All requested changes made @joboy-dev

**Pull request closed without merging**